winforms/src/toga_winforms/hardware/location.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from System import EventHandler
+from System.Device.Location import (
+    GeoCoordinate,
+    GeoCoordinateWatcher,
+    GeoPositionAccuracy,
+    GeoPositionChangedEventArgs,
+    GeoPositionPermission,
+)
+
+from toga import LatLng
+from toga.handlers import AsyncResult
+
+
+def toga_location(location: GeoCoordinate):
+    """Convert a GeoCoordinate into a Toga LatLng and altitude."""
+
+    if location.IsUnknown:
+        return None
+
+    return {
+        "location": LatLng(location.Latitude, location.Longitude),
+        "altitude": location.Altitude,
+    }
+
+
+class Location:
+    def __init__(self, interface):
+        self.interface = interface
+        self.watcher = GeoCoordinateWatcher(GeoPositionAccuracy.Default)
+        self._handler = EventHandler[GeoPositionChangedEventArgs[GeoCoordinate]](
+            self._position_changed
+        )
+        self._has_background_permission = False
+
+    def _position_changed(
+        self, sender, event: GeoPositionChangedEventArgs[GeoCoordinate]
+    ):
+        location = toga_location(event.Position.Location)
+        if location:
+            self.interface.on_change(**location)
+
+    def has_permission(self):
+        return self.watcher.Permission == GeoPositionPermission.Granted
+
+    def has_background_permission(self):
+        return self._has_background_permission
+
+    def request_permission(self, future: AsyncResult[bool]) -> None:
+        self.watcher.Start(False)  # TODO: where can we call stop?
+        future.set_result(self.has_permission())
```

Comment:
Is `Start()` a blocking method? A method that never returns False? My concern here is whether `has_permission()` is given a chance to return False on the basis of user input.

Reply:
I *believe* this overload of Start is blocking yes, but the docs aren't super clear, and it doesn't seem possible to clear the flag on the python exe to test locally properly